src/Controller/SiteInfoController.php

```diff
@@ -0,0 +1,203 @@
+      array_keys($form_display_widgets),
+      function ($collection, $field_name) use ($form_display_widgets, $field_definitions) {
+        $field_info = $form_display_widgets[$field_name];
+        if (!isset($field_info['region']) || $field_info['region'] !== 'content') {
+          return $collection;
+        }
+
+        if (isset($field_definitions[$field_name]) && $field_definitions[$field_name]->isRequired()) {
+          $collection[$field_name] = $field_info;
+        }
+
+        return $collection;
+      },
+      []
+    );
+
+    return $required_fields;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByCount(array $data) {
+    $bundles_by_count = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      if (!isset($bundle_info['instances'])) {
+        continue;
+      }
+
+      $bundles_by_count[$bundle_name] = $bundle_info['instances'];
+    }
+
+    arsort($bundles_by_count);
+
+    return $bundles_by_count;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByNumberOfFields(array $data) {
+    $bundles_by_number_of_fields = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      $bundles_by_number_of_fields[$bundle_name] = count(array_keys($bundle_info['fields']));
+    }
+    arsort($bundles_by_number_of_fields);
+
+    return $bundles_by_number_of_fields;
+  }
+
+  /**
+   * Gets site information.
+   *
+   * Allowed query options are following:
+   * => rule: {string} - options:
+   *   - count: to sort bundles by number of instance
+   *   - number_of_fields: to sort bundles by number of fields
+   * Default: "count"
+   *
+   * => index: {int}
+   * Default: 0
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The symfony request object.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   Returns JSON with site information needed by performance tests.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function siteInfo(Request $request) {
+    $rule = $request->query->get('rule', 'count');
+    $index = (int) $request->query->get('index', '0');
+
+    $data = (array) $this->cache()
+      ->get('thunder-performance-measurement:site-info:node');
+    if (!isset($data['data'])) {
+      $this->samplerMapping->enableMapping(FALSE);
+      $data = $this->samplerPluginManager->createInstance('bundle:node')->collect();
+      $data['bundles_by_count'] = $this->getBundlesByCount($data);
+      $data['bundles_by_number_of_fields'] = $this->getBundlesByNumberOfFields($data);
```

Comment:
The $data array contains the 'bundles_by_count' key at this point, which does not have the 'fields' values.
This throws some notices in the getBundlesByNumberOfFields() method. We either test, if the 'fields' key actually exists, or do not store the result of the getBundlesBy*() results in the data array.
I would vote for the latter and create a new variable which only contains the results of the getBundlesBy*() methods. This would also minimize the cached information.